homeassistant/components/climaveneta_imxw/__init__.py

```diff
@@ -0,0 +1,246 @@
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME, CONF_SLAVE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    ACTUAL_AIR_TEMPERATURE_REGISTER,
+    ACTUAL_WATER_TEMPERATURE_REGISTER,
+    ALARM_T1_REGISTER,
+    ALARM_T3_REGISTER,
+    ALARM_WATER_DRAIN_REGISTER,
+    CONF_HUB,
+    DOMAIN,
+    MODE_ON,
+    MODE_SUMMER,
+    MODE_WINTER,
+    PLATFORMS,
+    SCAN_INTERVAL,
+    STATE_READ_EV_WATER_REGISTER,
+    STATE_READ_FAN_AUTO_REGISTER,
+    STATE_READ_FAN_MAX_SPEED_REGISTER,
+    STATE_READ_FAN_MED_SPEED_REGISTER,
+    STATE_READ_FAN_MIN_SPEED_REGISTER,
+    STATE_READ_FAN_ONLY_REGISTER,
+    STATE_READ_ON_OFF_REGISTER,
+    STATE_READ_SEASON_REGISTER,
+    TARGET_TEMPERATURE_SUMMER_REGISTER,
+    TARGET_TEMPERATURE_WINTER_REGISTER,
+)
+
+# registers
+
+
+WATER_BYPASS = 0
+WATER_CIRCULATING = 1
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up platform from a ConfigEntry."""
+
+    hub = get_hub(hass, entry.data[CONF_HUB])
+    slave_id = entry.data[CONF_SLAVE]
+    name = entry.data[CONF_NAME]
+
+    result = await hub.async_pb_call(
+        slave_id, STATE_READ_ON_OFF_REGISTER, 1, CALL_TYPE_REGISTER_HOLDING
+    )
+    if result is None:
+        _LOGGER.error("Error reading value from Climaveneta iMXW modbus adapter")
+        return False
+
+    coordinator = ClimavenetaIMXWCoordinator(hass, hub, slave_id, name)
+    hass.data.setdefault(DOMAIN, {})
+    await coordinator.async_config_entry_first_refresh()
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Unload Climaveneta_imxw config entry."""
+
+    # our components don't have unload methods so no need to look at return values
+    for platform in PLATFORMS:
+        await hass.config_entries.async_forward_entry_unload(config_entry, platform)
+
+    return True
+
+
+class ClimavenetaIMXWCoordinator(DataUpdateCoordinator):
+    """My custom coordinator."""
+
+    fan_mode = FAN_AUTO
+    hvac_mode = HVACMode.OFF
+    hvac_action = HVACAction.OFF
+
+    def __init__(self, hass: HomeAssistant, hub, slaveid, name) -> None:
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            # Name of the data. For logging purposes.
+            name=name,
+            # Polling interval. Will only be polled if there are subscribers.
+            update_interval=SCAN_INTERVAL,
+        )
+        self.hub = hub
+        self.slave_id = slaveid
+        self.data_modbus = {"a": 1}
+        self.name = name
+
+    async def _async_update_data(self):
```

Comment:
Right, this integration doesn't use a library to connect with an external device as required in https://developers.home-assistant.io/docs/creating_platform_code_review#5-communication-with-devicesservices. Although I am not sure how this works together with modbus, so I already asked someone if he knows an example for you to look at.

Reply:
This integration was largely based on Flexit integration.